stakewise_cli/commands/create_referrals_proposal.py

```diff
@@ -0,0 +1,165 @@
+import decimal
+
+import click
+from web3 import Web3
+
+from stakewise_cli.coingecko import get_average_range_price
+from stakewise_cli.eth1 import get_block_timestamp, get_referrals, get_web3_client
+from stakewise_cli.ipfs import upload_to_ipfs
+from stakewise_cli.networks import GNOSIS_CHAIN, GOERLI, MAINNET, NETWORKS, PERM_GOERLI
+from stakewise_cli.proposals import generate_referrals_swise_specification
+from stakewise_cli.queries import get_ethereum_gql_client, get_stakewise_gql_client
+
+w3 = Web3()
+
+
+@click.command(
+    help="Creates referral proposal and generates a forum post specification"
+)
+@click.option(
+    "--network",
+    default=MAINNET,
+    help="The network to generate the referral proposal for",
+    prompt="Enter the network name",
+    type=click.Choice(
+        [MAINNET, GOERLI, PERM_GOERLI, GNOSIS_CHAIN], case_sensitive=False
+    ),
+)
+@click.option(
+    "--from-block",
+    required=True,
+    help="The block number to start collecting referrals from",
+    prompt="Enter the block number to start collecting referrals from",
+    type=int,
+)
+@click.option(
+    "--to-block",
+    required=True,
+    help="The block number to collect referrals up to",
+    prompt="Enter the block number to collect referrals up to",
+    type=int,
+)
+@click.option(
+    "--referrals-share",
+    required=True,
+    prompt="Enter referrals share (%), ex. 1.5 = 1.5%",
+    help="Referrals share (%), ex. 1.5 = 1.5%",
+    type=decimal.Decimal,
+)
+@click.option(
+    "--swise-price",
+    help="Average SWISE price in USD for the period",
+    type=decimal.Decimal,
+)
+@click.option(
+    "--eth-price",
+    help="Average ETH price in USD for the period",
+    type=decimal.Decimal,
+)
+@click.option(
+    "--whitelist-path",
+    help="The file path from where to read whitelisted accounts",
+    prompt="Enter the file path from where to read whitelisted accounts. One address per line",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+def create_referrals_proposal(
+    network: str,
+    from_block: int,
+    to_block: int,
+    referrals_share: decimal.Decimal,
+    swise_price: decimal.Decimal,
+    eth_price: decimal.Decimal,
+    whitelist_path: str,
+) -> None:
+    if not from_block < to_block:
+        click.echo(
```

Comment:
Here and in other places I suggest we use `raise click.ClickException` instead of echo + return